File: app/src/main/java/com/petnbu/petnbu/repo/LoadMoreState.java
package com.petnbu.petnbu.repo;

public class LoadMoreState {
    private final boolean running;
    private final String errorMessage;
    private boolean handledError = false;

    public LoadMoreState(boolean running, String errorMessage) {
        this.running = running;
        this.errorMessage = errorMessage;
    }

    public boolean isRunning() {
        return running;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public String getErrorMessageIfNotHandled() {
        if (handledError) {
            return null;
        }
        handledError = true;
        return errorMessage;
    }

    @Override
    public String toString() {
        return "LoadMoreState{" +
                "running=" + running +
                ", errorMessage='" + errorMessage + '\'' +
                ", handledError=" + handledError +
                '}';
    }
}

File: app/src/main/java/com/petnbu/petnbu/repo/LoadMoreState.kt
package com.petnbu.petnbu.repo

class LoadMoreState(val isRunning: Boolean, val errorMessage: String?) {
    private var handledError = false

    val errorMessageIfNotHandled: String?
        get() {
            if (handledError) {
                return null
            }
            handledError = true
            return errorMessage
        }

    override fun toString(): String {
        return "LoadMoreState{" +
                "running=" + isRunning +
                ", errorMessage='" + errorMessage + '\''.toString() +
                ", handledError=" + handledError +
                '}'.toString()
    }
}
